src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -0,0 +1,237 @@
+      CatchMobileCommandResult(CallOnTimeOut(*command)));
+  EXPECT_EQ(sended_result_msg, received_result_msg);
+}
+
+TEST_F(AlertRequestTest, Run_ApplicationIsNotRegistered_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  ON_CALL(app_mngr_, application(_))
+      .WillByDefault(Return(ApplicationSharedPtr()));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::APPLICATION_NOT_REGISTERED,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_AlertFrequencyIsTooHigh_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(true));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::REJECTED,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_FailToProcessSoftButtons_UNSUCCESS) {
+  const mobile_apis::Result::eType kResultCode =
+      mobile_apis::Result::INVALID_ENUM;
+
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
+  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+              ProcessSoftButtons(_, _, _, _)).WillOnce(Return(kResultCode));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(kResultCode,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(AlertRequestTest, Run_MandatoryParametersAreMissed_UNSUCCESS) {
+  CommandPtr command(CreateCommand<AlertRequest>());
+
+  MockAppPtr app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
+  ON_CALL(*app, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+  ON_CALL(*app, AreCommandLimitsExceeded(_, _)).WillByDefault(Return(false));
+
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(
+          ReturnRef(*static_cast<policy::PolicyHandlerInterface*>(NULL)));
```

Comment:
@okozlovlux This is the third dereferenced NULL pointer I encounter in this pull request. According to the C++ standard this leads to undefined behavior. So you definitely have to remove them.